components/raftstore/src/store/fsm/peer.rs

```diff
@@ -3115,6 +3115,14 @@ where
             return Ok(());
         }
 
+        if self.fsm.peer.in_joint_state() {
```

Comment:
I think you should only prevent prepare merge otherwise it can be an incompatible change.